### PR TITLE
Update database.py

### PR DESCRIPTION
**Pull request by AndrewDorn**
Updated docs in database.py

## Summary of Changes

Added some doc strings to database.py

## How Has This Been Tested?

Non-functional changes.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
    - [ ] Unit tests
    - [ ] Integration tests
    - [ ] Manual tests (outlined above)
- [x] New and existing unit tests pass locally with my changes
    - [ ] Automated tests passing (if not fork)
    - [ ] Manual testing/integration testing passes (if forked)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

File: tasks/pg_utils/database.py [hunk 1]
```diff
@@ -11,93 +11,135 @@
...
+
+    Returns: todo
     """
-    json_result = json.loads(json.dumps(result_rows, default=myconverter))
+    json_result = json.loads(json.dumps(result_rows, default=myconverter))  # todo: wha?
     return json_result
 
 
-def myconverter(obj):       #pylint: disable-msg=inconsistent-return-statements
+# todo: Figure out how the README.md is generated.
```
Comment (ifestus) [inline tasks/pg_utils/database.py:78]: Which `README.md` are you talking about here?

File: tasks/pg_utils/database.py [hunk 2]
```diff
@@ -183,12 +232,16 @@ def read_db_connect_info(param_source):
     return dbconnect_info
 
 
-def get_db_connect_info(env_or_ssm, param_name, decrypt=False):
-    """
+def get_db_connect_info(env_or_ssm: str, param_name, decrypt: bool = False):
```
Comment (ifestus) [inline tasks/pg_utils/database.py:235]: Should there be a `-> <type>` in this function definition?

File: tasks/pg_utils/database.py [hunk 3]
```diff
@@ -113,14 +155,21 @@ def get_cursor(dbconnect_info):
             cursor.close()
 
 
-
-def single_query(sql_stmt, dbconnect_info, params=None):
+def single_query(sql_stmt, dbconnect_info: dict[str, Union[str, int]], params=None):
```
Comment (ifestus) [inline tasks/pg_utils/database.py:158]: Should there be a -> <type> in this function definition?

**Review by ifestus — APPROVED** [review]
I have a few questions, but otherwise LGTM :+1:

File: tasks/pg_utils/database.py (hunk 3 above)
Comment (AndrewDorn) [reply tasks/pg_utils/database.py]: There should be, but I'm a bit unsure of the type. Added what I could.

File: tasks/pg_utils/database.py (hunk 1 above)
Comment (AndrewDorn) [reply tasks/pg_utils/database.py:77]: The one in pg_utils. Contains documentation on each function, so I'm not sure if I'd have to manually update it or if it's generated in some way,

File: tasks/pg_utils/database.py (hunk 2 above)
Comment (AndrewDorn) [reply tasks/pg_utils/database.py:233]: Fixed

**Pull request merged**